File: src/main/java/org/fletchly/genius/ollama/model/OllamaOptions.java
package org.fletchly.genius.ollama.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;

@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public record OllamaOptions(
        double temperature,
        @JsonProperty("top_k")
        int topK,
        @JsonProperty("top_p")
        double topP,
        @JsonProperty("num_predict")
        int numPredict
) {
}

File: src/main/kotlin/io/fletchly/genius/ollama/model/OllamaOptions.kt
package io.fletchly.genius.ollama.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class OllamaOptions(
    val temperature: Double,
    @SerialName("top_k")
    val topK: Int,
    @SerialName("top_p")
    val topP: Double,
    @SerialName("num_predict")
    val numPredict: Int
)
